www/src/hooks/use-active-hash.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react"
+
+export const useActiveHash = (itemIds, rootMargin = undefined) => {
+  const [activeHash, setActiveHash] = useState(``)
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
+      entries => {
+        entries.forEach(entry => {
+          if (entry.isIntersecting) {
+            setActiveHash(entry.target.id)
+          }
+        })
+      },
+      { rootMargin: rootMargin || `0% 0% -80% 0%` }
+    )
+
+    itemIds.forEach(id => {
+      observer.observe(document.getElementById(id))
+    })
+
+    return () => {
+      itemIds.forEach(id => {
+        observer.unobserve(document.getElementById(id))
+      })
+    }
+  }, [])
+
+  useEffect(() => {
+    if (activeHash) {
+      window.history.replaceState(null, null, `#${activeHash}`)
```

Comment:
I'm not sure I'm totally onboard with this behavior 🤔  I know this was mentioned in the issue for this change so I'll cc in @tesseralis,though we don't currently update the hash in the tutorial as you scroll through it and I think there are some pros and cons both ways.

This means users sharing links to docs are going to almost always have a hash in them unless they delete it themselves. These have a tendency to get stale since the titles can change pretty often. A stale hash means the link will just link to the top of the page which isn't a big deal, but I bet people will share links and end up deep linking halfway down the page when it might be better to default to start from the top of a guide for context. Our guides don't tend to be super long and we provide the ability to click a link by the headers to add the hash to the url. Interested in other thoughts.

Reply:
Yeah, I buy that. I think if we were to implement this feature it would *only* be for pages that are so long that they're essentially multiple pages -- for example, the recipes page before we split it up. I did a quick audit of a couple of doc sites and I don't think it's a common pattern, so I'm okay with reverting back to our current behavior (not updating the hash).

Reply:
Okay! Fixed in 53399bb